### PR TITLE
Add each precourse challenges into precourse folder

### PR DESCRIPTION
**Pull request by martin-garcia-blanco**

### CODE REVIEW

File: staff/martin-garcia/staff/Airlines/index.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    });
+
+    console.log("\nLos últimos vuelos del día son: "); //Últimos 5 vuelos del día
+    ultimosVuelos = flights.slice(-5);
+
+    ultimosVuelos.forEach(function(element) {
+        console.log(element.from + " - " + element.to + " precio:" + element.cost + "€");
+    })
+
+}
```
Comment (adria-zhu) [inline staff/martin-garcia/staff/Airlines/index.js:49]: Missing Semicolon

File: staff/martin-garcia/staff/Airlines/index.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    flights.forEach(function(element) { //Vuelos con escalas
+        if (element.scale) console.log(element.from + " - " + element.to + " precio:" + element.cost + "€");
+    });
+
+    console.log("\nLos últimos vuelos del día son: "); //Últimos 5 vuelos del día
+    ultimosVuelos = flights.slice(-5);
+
+    ultimosVuelos.forEach(function(element) {
+        console.log(element.from + " - " + element.to + " precio:" + element.cost + "€");
+    })
```
Comment (adria-zhu) [inline staff/martin-garcia/staff/Airlines/index.js]: Missing Semicolon

File: staff/martin-garcia/staff/AirlinesPro/index.js [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+                if (element.id == idVuelo) {
+                    validacionId = true;
+                    return;
+                }
+            });
+
+            validacionId ? console.log("Gracias por su compra, vuelva pronto") : console.log("No se ha entoncreado su vuelo");
+
+        } else {
+            console.log("Hasta la próxima")
```
Comment (adria-zhu) [inline staff/martin-garcia/staff/AirlinesPro/index.js]: Missing Semicolon

File: staff/martin-garcia/staff/Bingo Pro/bingoPro.js [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+    var carton;
+    do {
+        var validacion = true;
+        carton = generadorBingoCard(numero);
+        var cartonStr = "";
+        for (var i = 0; i < carton.length; i++) {
+            for (var x = 0; x < carton[i].length; x++) {
+                cartonStr += " " + carton[i][x].number;
+            }
+            cartonStr += "\n"
```
Comment (adria-zhu) [inline staff/martin-garcia/staff/Bingo Pro/bingoPro.js]: Missing Semicolon

File: staff/martin-garcia/staff/PasapalabraHTML/style.css [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    color: black;
+}
+
+.boton {
+    color: white;
+    background-color: #3e64ff;
+    font-family: 'Manjari', sans-serif;
+    font-size: 20px;
+    padding: 5px;
+    font-weight: bold
```
Comment (adria-zhu) [inline staff/martin-garcia/staff/PasapalabraHTML/style.css:54]: Missing semicolon

File: staff/martin-garcia/staff/AirlinesPro/index.js [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+            var tamaño = flights.length;
+            for (var a = 0; a < flights.length; a++) {
+                if (flights[a].id == idVuelo) {
+                    flights.splice(a, 1);
+                    console.log("vuelo con id " + idVuelo + " eliminado.");
+                }
+            }
+            if (tamaño == flights.length) console.log("No se encontró el vuelo con el id " + idVuelo);
+        } else {
+            console.log("Hasta la próxima")
```
Comment (adria-zhu) [inline staff/martin-garcia/staff/AirlinesPro/index.js]: lack of semicolon

File: staff/martin-garcia/staff/Bingo Pro/bingoPro.js [hunk 7]
```diff
@@ -0,0 +1,234 @@
+//Los números estaran comprendidos en el rango [1,99]
+
+
+bingo();
+
+function bingo() {
+    var cantidadNumeros = 15; //Cantidad de números en el cartón
+    var ranking = []; //Array de objetos {nombre:"Pepe", puntuacion:15}
+    var validaPartida = false; //esta variable llevará el valor de la función confirm();
+    mostrarRanking(ranking, cantidadNumeros)
```
Comment (adria-zhu) [inline staff/martin-garcia/staff/Bingo Pro/bingoPro.js]: lack of semicolon

File: staff/martin-garcia/staff/Airlines/index.js (hunk 1 above)
Comment (martin-garcia-blanco) [reply staff/martin-garcia/staff/Airlines/index.js:49]: Solved

File: staff/martin-garcia/staff/Bingo Pro/bingoPro.js [hunk 8]
```diff
@@ -0,0 +1,234 @@
...
+        for (var i = 0; i < bingoCard.length; i++) {
+            for (var x = 0; x < bingoCard[i].length; x++) {
+                cartonStr += " " + bingoCard[i][x].number;
+                if (bingoCard[i][x].mathched) {
+                    cartonModificado += " X";
+                } else {
+                    cartonModificado += " " + bingoCard[i][x].number;
+                }
+            }
+            cartonStr += "\n"
```
Comment (adria-zhu) [inline staff/martin-garcia/staff/Bingo Pro/bingoPro.js:223]: laack of semicolon

File: staff/martin-garcia/staff/AirlinesPro/index.js (hunk 3 above)
Comment (martin-garcia-blanco) [reply staff/martin-garcia/staff/AirlinesPro/index.js]: solved

File: staff/martin-garcia/staff/Bingo Pro/bingoPro.js (hunk 4 above)
Comment (martin-garcia-blanco) [reply staff/martin-garcia/staff/Bingo Pro/bingoPro.js]: solved

File: staff/martin-garcia/staff/PasapalabraHTML/work.js [hunk 9]
```diff
@@ -0,0 +1,298 @@
...
+
+
+//El tipo varía entre primerNombre, nuevoNombre, dependiendo de si se ha pedido ingresar un nuevo nombre
+function start(tipo) {
+    if (tipo === "primerNombre") {
+        jugador.nombre = document.getElementById("nombre").value;
+    } else {
+        var segundoNombre = document.getElementById("segundoNombre");
+        if (segundoNombre.value.length > 0) {
+            jugador.nombre = segundoNombre.value
```
Comment (adria-zhu) [inline staff/martin-garcia/staff/PasapalabraHTML/work.js]: lack of semicolon

File: staff/martin-garcia/staff/PasapalabraHTML/style.css (hunk 5 above)
Comment (martin-garcia-blanco) [reply staff/martin-garcia/staff/PasapalabraHTML/style.css:54]: solved

File: staff/martin-garcia/staff/PasapalabraHTML/work.js [hunk 10]
```diff
@@ -0,0 +1,298 @@
...
+    timer(); //Inicia el temporizador del juego
+    formularPregunta("PrimeraRonda");
+}
+
+
+
+//Resetea los datos del juego para poder comenzar de cero.
+function resetGame() {
+    document.getElementById("comprobacionPregunta").innerHTML = "";
+    //document.getElementById("comprobacionRespuesta").innerHTML = "";
```
Comment (adria-zhu) [inline staff/martin-garcia/staff/PasapalabraHTML/work.js]: Unecessary code

File: staff/martin-garcia/staff/calculatorHTML/work.js [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+var operadorSTR = "";
+var operadores = "X+-/";
+
+function añadirNumero(e) {
+    actualizarDisplay(e.target.innerText);
+}
+
+
+
+/* function resultado(e) {
```
Comment (adria-zhu) [inline staff/martin-garcia/staff/calculatorHTML/work.js]: unnecessary code

File: staff/martin-garcia/staff/calculatorHTML/work.js [hunk 12]
```diff
@@ -0,0 +1,136 @@
...
+    textoDisplay = "";
+    document.getElementsByClassName("display")[0].innerHTML = "0";
+}
+
+
+//si el primer número es error, no se permite clicar en operandores
+function operador(e) {
+    var display = document.getElementsByClassName("display");
+    if (display[0].innerHTML == "NaN" || display[0].innerHTML == "Infinity" || display[0].innerHTML == "-Infinity") {
+        return
```
Comment (adria-zhu) [inline staff/martin-garcia/staff/calculatorHTML/work.js:67]: semicolon

File: staff/martin-garcia/staff/PasapalabraHTML/work.js [hunk 13]
```diff
@@ -0,0 +1,298 @@
...
+// inicio: antes de clicar          jugando: Mientras se pregunta           finalizado: Una vez se termina la partida
+
+
+//Se utilizará el status para saber si está:
+//sin contestar 0           acertada 1          fallada -1
+//Para que el juego fuese más interesante he añadido 5 preguntas por letra, para que puedan salir diferentes definiciones
+
+var noContestadas = [];
+var questions = [];
+//var val;
```
Comment (adria-zhu) [inline staff/martin-garcia/staff/PasapalabraHTML/work.js]: unnecessary code

File: staff/martin-garcia/staff/PasapalabraHTML/work.js [hunk 14]
```diff
@@ -0,0 +1,298 @@
...
+            falladas += 1;
+        }
+    });
+    ranking.push({ nombre: jugador.nombre, aciertos: acertadas, fallos: falladas });
+}
+
+
+//Muestra el ranking por pantalla y pregunta si se quiere jugar otra vez
+function mostrarRanking() {
+    //var ranking = document.getElementById("ranking");
```
Comment (adria-zhu) [inline staff/martin-garcia/staff/PasapalabraHTML/work.js]: unnecessary code

File: staff/martin-garcia/staff/Bingo Pro/bingoPro.js [hunk 15]
```diff
@@ -0,0 +1,234 @@
...
+        var respuesta = prompt("Si desea jugar con este cartón, escriba 'Yes'," +
+            " pulse intro para cambiar de carton\n" +
+            cartonStr);
+        if (respuesta == null || (respuesta.toUpperCase()).localeCompare("YES")) {
+            validacion = false;
+        }
+    } while (!validacion);
+    return carton;
+}
+//Nuevo turno
```
Comment (adria-zhu) [inline staff/martin-garcia/staff/Bingo Pro/bingoPro.js:174]: unnecessary code

File: staff/martin-garcia/staff/Bingo Pro/bingoPro.js [hunk 16]
```diff
@@ -0,0 +1,234 @@
...
+            aux.push(linea);
+            linea = [];
+        } else if (i == (number - 1)) {
+            aux.push(linea);
+        }
+    }
+    return aux;
+}
+
+//Genera un n aleaorio en el rango [1,99]
```
Comment (AdriaanZhu) [inline staff/martin-garcia/staff/Bingo Pro/bingoPro.js]: unecessary code

**Review by adria-zhu — COMMENTED** [review]
most of the problem were unnecessary code

File: staff/martin-garcia/staff/AirlinesPro/index.js (hunk 6 above)
Comment (martin-garcia-blanco) [reply staff/martin-garcia/staff/AirlinesPro/index.js]: done

File: staff/martin-garcia/staff/PasapalabraHTML/work.js [hunk 17]
```diff
@@ -0,0 +1,298 @@
...
+        } else if (noContestadas[numeroPreguntaNoContestada].status == -1) {
+            console.log("False");
+            questions[questions.indexOf(noContestadas[numeroPreguntaNoContestada])].status == -1;
+            noContestadas.splice(numeroPreguntaNoContestada, 1);
+            numeroPreguntaNoContestada -= 1;
+        } else {
+            numeroPreguntaNoContestada += 1;
+        }
+    }
+    validacion.contadorVuelta += 1;
```
Comment (jmrodri610) [inline staff/martin-garcia/staff/PasapalabraHTML/work.js:158]: This function can be more simple with a switch case :)

File: staff/martin-garcia/staff/Bingo Pro/bingoPro.js (hunk 8 above)
Comment (martin-garcia-blanco) [reply staff/martin-garcia/staff/Bingo Pro/bingoPro.js:223]: done

File: staff/martin-garcia/staff/PasapalabraHTML/work.js (hunk 9 above)
Comment (martin-garcia-blanco) [reply staff/martin-garcia/staff/PasapalabraHTML/work.js]: done

File: staff/martin-garcia/staff/PasapalabraHTML/work.js (hunk 10 above)
Comment (martin-garcia-blanco) [reply staff/martin-garcia/staff/PasapalabraHTML/work.js]: done

File: staff/martin-garcia/staff/PasapalabraHTML/work.js (hunk 13 above)
Comment (martin-garcia-blanco) [reply staff/martin-garcia/staff/PasapalabraHTML/work.js]: done

File: staff/martin-garcia/staff/PasapalabraHTML/work.js (hunk 14 above)
Comment (martin-garcia-blanco) [reply staff/martin-garcia/staff/PasapalabraHTML/work.js]: solved

File: staff/martin-garcia/staff/Bingo Pro/bingoPro.js [hunk 18]
```diff
@@ -0,0 +1,234 @@
...
+//var numeroBombo = () => { return Math.round((Math.random() * (99 - 1 + 1)) + 1); }
```
Comment (adria-zhu) [inline staff/martin-garcia/staff/Bingo Pro/bingoPro.js]: unecessary code

File: staff/martin-garcia/staff/PasapalabraHTML/work.js [hunk 19]
```diff
@@ -0,0 +1,298 @@
...
+    preguntasRosco();
+    if (validacion.respuesta == false) {
+        addToRanking();
+        //en esta funcion se pregunta mediante un confirm si se quiere jugar de nuevo
+        //validacion.partida = mostrarRanking();
```
Comment (adria-zhu) [inline staff/martin-garcia/staff/PasapalabraHTML/work.js]: unecessary code

File: staff/martin-garcia/staff/PasapalabraHTML/work.js [hunk 20]
```diff
@@ -0,0 +1,298 @@
...
+        mostrarRanking();
+        cambiarDescripcion("reiniciar");
+    }
+    if (validacion.partida == false) {
+        //Nunca chega aquí
```
Comment (adria-zhu) [inline staff/martin-garcia/staff/PasapalabraHTML/work.js]: unnecessary code

File: staff/martin-garcia/staff/calculatorHTML/work.js (hunk 11 above)
Comment (martin-garcia-blanco) [reply staff/martin-garcia/staff/calculatorHTML/work.js]: erased

File: staff/martin-garcia/staff/Bingo Pro/bingoPro.js (hunk 18 above)
Comment (martin-garcia-blanco) [reply staff/martin-garcia/staff/Bingo Pro/bingoPro.js]: done

File: staff/martin-garcia/staff/PasapalabraHTML/work.js (hunk 19 above)
Comment (martin-garcia-blanco) [reply staff/martin-garcia/staff/PasapalabraHTML/work.js]: done, thx

File: staff/martin-garcia/staff/PasapalabraHTML/work.js (hunk 20 above)
Comment (martin-garcia-blanco) [reply staff/martin-garcia/staff/PasapalabraHTML/work.js]: done, thx

File: staff/martin-garcia/staff/PasapalabraHTML/work.js (hunk 17 above)
Comment (martin-garcia-blanco) [reply staff/martin-garcia/staff/PasapalabraHTML/work.js:158]: Yes, thx for the tip, It's already solved

**Pull request merged**